src/Signup.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import axios from "axios";
+import history from './history';
+import './todo.css';
+
+class Signup extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            username: "",
+            password: "",
+            confirmPassword: "",
+            cred: {}
+        };
+    }
+
+    setUsername = event => {
+        let value = event.target.value;
+        this.setState({
+            username: value
+        });
+    };
+
+    setPassword = event => {
+        let value = event.target.value;
+        this.setState({
+            password: value
+        });
+    };
+
+    setConfirmPassword = event => {
+        let value = event.target.value;
+        this.setState({
+            confirmPassword: value
+        });
+    }
+
+    signUp = () => {
+        if (this.state.username === '' || this.state.password === '') {
+            alert('Fields can not be empty');
+        }
+        else if (this.state.password.length > 5) {
```

Comment:
handling validation on value entering/changing on input tag would be best practice